### ISSUE TITLE
Run tests against mock API 

### ISSUE THREAD

**Issue by muety** [jaebradley/wakatime-client#103]
While @jaebradley is the only contributor to this project, it's fine to run tests against the actual WakaTime API using his account and credentials. However, in the future, it might be beneficial to stub the endpoints to make tests more predictable. E.g. `WakaTimeClient Integration Test › getUser › gets user details` expects a specific user object and therefore will fail when run against a different WakaTime account. Also, it feels somehow odd to run tests against the production web service.

I'm not too involved in the JavaScript ecosystem, so I don't have any concrete proposals on how to best realize this. In the Java world, tools like [WireMock](http://wiremock.org/) exist. Probably there are (maybe more lightweight) alternative for Node projects as well.

**Comment by jaebradley** [comment]
I agree - I've been lazy about setting up a better system that doesn't have values hard-coded to mine.

> Also, it feels somewhat odd to run tests against the production web service.

I see the point, though this is kind've the "realest" way to test against a system I have no affiliation with nor have knowledge / access to changes that have been made internally that might affect the client's output.
